lnst/Recipes/ENRT/ConfigMixins/BaseRESTConfigMixin.py

```diff
@@ -28,11 +34,25 @@ def __build_request(self, endpoint: str, **kwargs):
 
         return kwargs
 
-    def api_request(self, method: str, endpoint: str, response_code: int = 200, **kwargs) -> bytes:
+    def api_request(self, method: str, endpoint: str, response_code: int = 200, tries: int = 7, **kwargs) -> bytes:
         request = self.__build_request(endpoint, **kwargs)
         req_func = self.__get_request_function(method)
 
-        response = req_func(**request)
+        for i in range(1, tries+1):  # i starts from 1
+            try:
+                response = req_func(**request)
+            except RequestException as e:
+                delay = self.BASE_DELAY * 2 ** i
+
+                logging.error(f"API request ({i}/{tries}) failed: {e}. Retrying in {delay} seconds.")
```

Comment:
ah, sorry, one more thing. please add a `log_exception_traceback` here... the simple `str(e)` part can sometimes be quite useless but a log of the traceback can help with that.